src/Raven.Server/RavenServer.cs

```diff
@@ -2734,8 +2754,90 @@ internal class DebugPackageTestingStuff
+            {
+                throw new InvalidOperationException(
+                    $"The configuration option '{RavenConfiguration.GetKey(x => x.Security.WellKnownIssuerHashes)}' has been deprecated and should not be used. You should instead use '{RavenConfiguration.GetKey(x => x.Security.WellKnownIssuers)}'.");
+            }
+            if (Configuration.Security.WellKnownIssuers == null)
+                return;
+
+            WellKnownIssuersThumbprints = new string[Configuration.Security.WellKnownIssuers.Length];
+            WellKnownIssuers = new X509Certificate2[Configuration.Security.WellKnownIssuers.Length];
+            byte[] buffer = ArrayPool<byte>.Shared.Rent(8192);
+            for (int index = 0; index < Configuration.Security.WellKnownIssuers.Length; index++)
+            {
+                string issuer = Configuration.Security.WellKnownIssuers[index];
+                if (issuer.Length > buffer.Length) // the rate is actually 75%, but easier to just assume 1:1 here, we have enough space
+                {
+                    ArrayPool<byte>.Shared.Return(buffer);
+                    buffer = ArrayPool<byte>.Shared.Rent(issuer.Length);
+                }
+
+                X509Certificate2 certificate;
+                if (Convert.TryFromBase64String(issuer, buffer, out var read))
+                {
+                    try
+                    {
+                        certificate = new X509Certificate2(buffer[0..read]);
+                    }
+                    catch (Exception e)
+                    {
+                        throw new InvalidOperationException(
+                            $"Unable to parse the provided '{RavenConfiguration.GetKey(x => x.Security.WellKnownIssuers)}' value: {issuer[..Math.Min(64, issuer.Length)]}",
+                            e);
+                    }
+                }
+                else // maybe it's a path?
+                {
+                    try
+                    {
+                        certificate = new X509Certificate2(issuer);
+                    }
+                    catch (Exception e)
+                    {
+                        throw new InvalidOperationException($"Unable to read file provided via '{RavenConfiguration.GetKey(x => x.Security.WellKnownIssuers)}' from: {issuer}",
+                            e);
+                    }
+                }
+
+                if (certificate.HasPrivateKey)
+                    throw new InvalidOperationException(
+                        $"The certificate provided by '{RavenConfiguration.GetKey(x => x.Security.WellKnownIssuers)}' configuration for {certificate.SubjectName} {certificate.Thumbprint} includes the PRIVATE KEY, that is not a secured model, and was rejected by RavenDB");
+                
+                WellKnownIssuers[index] = certificate;
+                WellKnownIssuersThumbprints[index] = certificate.Thumbprint;
+            }
 
+            ArrayPool<byte>.Shared.Return(buffer);
```

Comment:
Should we put that in finally block?

Reply:
No, if this throws, it is going to be GCed anyway. And the cost of `try/finally` is actually higher in tight spaces.